src/System.IO.FileSystem/src/System/IO/UnixFileStream.cs

```diff
@@ -1284,24 +1284,16 @@ private long SeekCore(long offset, SeekOrigin origin)
             Debug.Assert(handle != null);
             Debug.Assert(!handle.IsInvalid);
 
-            // System calls may fail due to EINTR (signal interruption).  We need to retry in those cases.
-            while (true)
+            long result = sysCall(handle, arg1, arg2);
```

Comment:
I've changed the checking to a `CheckFileCall` method, and removed the delegate.